### ISSUE TITLE
[feature request]make cache expand expand-copy-close way?

### ISSUE THREAD

**Issue by makeyang** [elastic/elasticsearch#21086]
**Describe the feature**:
currently we I expand(enlarge the memory) filter cache, the action is to close th old one and build a new one, which will cause previously cache completely removed.
can it be made to expand-copy-close way?

**Comment by jpountz** [comment]
Actually we removed the ability to resize the filter cache has been removed for quite some time now. Like you noticed it had the bad side effect of clearing the cache. It was also quite a tricky operation in terms of concurrency: the implementation was not 100% correct, which could lead to memory leaks in some rare cases. So we decided to not offer the ability to resize the filter cache live when we switched to a new cache. This new requires nodes to be restarted.
